dbms/src/Storages/DeltaMerge/File/DMFilePackFilter.h

```diff
@@ -284,10 +290,17 @@ class DMFilePackFilter
         if (!dmfile->isColIndexExist(col_id))
             return;
 
+        Stopwatch watch;
         loadIndex(param.indexes, dmfile, file_provider, index_cache, set_cache_if_miss, col_id, read_limiter);
+
+        if (dm_context && dm_context->table_scan_context_ptr)
```

Comment:
Can we enforce dm_context and dm_context->table_scan_context_ptr are always present? Then we can avoid writing `if (...)` everywhere.